icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/DateIntervalFormatTest.java

```diff
@@ -1889,4 +1890,60 @@ public void run() {
             }
         }
     }
+
+    @Test
+    public void testFormattedDateInterval() {
+        DateIntervalFormat fmt = DateIntervalFormat.getInstance("dMMMMy", ULocale.US);
+
+        {
+            String message = "FormattedDateInterval test 1";
+            String expectedString = "July 20 \u2013 25, 2018";
+            Calendar input1 = Calendar.getInstance(ULocale.UK);
+            Calendar input2 = Calendar.getInstance(ULocale.UK);
+            input1.set(2018, 6, 20);
+            input2.set(2018, 6, 25);
+            FormattedDateInterval result = fmt.formatToValue(input1, input2);
+            Object[][] expectedFieldPositions = new Object[][] {
+                // field, begin index, end index
+                {DateFormat.Field.MONTH, 0, 4},
+                {DateIntervalFormat.SpanField.DATE_INTERVAL_SPAN, 5, 7, 0},
+                {DateFormat.Field.DAY_OF_MONTH, 5, 7},
+                {DateIntervalFormat.SpanField.DATE_INTERVAL_SPAN, 10, 12, 1},
+                {DateFormat.Field.DAY_OF_MONTH, 10, 12},
+                {DateFormat.Field.YEAR, 14, 18}};
+            FormattedValueTest.checkFormattedValue(
+                message,
+                result,
+                expectedString,
+                expectedFieldPositions);
+        }
+
+        // To test the fallback pattern behavior, make a custom DateIntervalInfo.
+        DateIntervalInfo dtitvinf = new DateIntervalInfo();
+        dtitvinf.setFallbackIntervalPattern("<< {1} --- {0} >>");
+        fmt = DateIntervalFormat.getInstance("dMMMMy", ULocale.US, dtitvinf);
+
+        {
+            String message = "FormattedDateInterval with fallback format test 1";
+            String expectedString = "<< July 25, 2018 --- July 20, 2018 >>";
+            Date input1 = new Date(2018 - 1900, 6, 20);
+            Date input2 = new Date(2018 - 1900, 6, 25);
+            DateInterval dtiv = new DateInterval(input1.getTime(), input2.getTime());
+            FormattedDateInterval result = fmt.formatToValue(dtiv);
+            Object[][] expectedFieldPositions = new Object[][] {
+                {DateIntervalFormat.SpanField.DATE_INTERVAL_SPAN, 3, 16, 1},
+                {DateFormat.Field.MONTH, 3, 7},
+                {DateFormat.Field.DAY_OF_MONTH, 8, 10},
+                {DateFormat.Field.YEAR, 12, 16},
+                {DateIntervalFormat.SpanField.DATE_INTERVAL_SPAN, 21, 34, 0},
+                {DateFormat.Field.MONTH, 21, 25},
+                {DateFormat.Field.DAY_OF_MONTH, 26, 28},
+                {DateFormat.Field.YEAR, 30, 34}};
+            FormattedValueTest.checkFormattedValue(
+                message,
+                result,
+                expectedString,
+                expectedFieldPositions);
+        }
```

Comment:
Could you add a test case for the time-only fallback pattern? (e.g. "Feb 15, 2 PM – 3 PM")?

Reply:
Done.